pkg/components/server.go

```diff
@@ -421,6 +446,94 @@ func (s *serverImpl) rebuildStatefulSet() *appsv1.StatefulSet {
+	logsDeliveryPath := "//sys/admin/logs"
+	if commonSpec.BuiltinSidecars != nil && commonSpec.BuiltinSidecars.LogsDeliveryPath != "" {
+		logsDeliveryPath = commonSpec.BuiltinSidecars.LogsDeliveryPath
+	}
+	if instanceSpec.BuiltinSidecars != nil && instanceSpec.BuiltinSidecars.LogsDeliveryPath != "" {
+		logsDeliveryPath = instanceSpec.BuiltinSidecars.LogsDeliveryPath
+	}
+	return logsDeliveryPath
+}
+
+func extractTimbertruckImageFromSpec(commonSpec ytv1.CommonSpec, instanceSpec *ytv1.InstanceSpec) string {
+	// If none of the loggers are going to deliver logs to cypress, then don't return the image.
+	// This is important because otherwise the StatefulSet will wait for a timbertruck to be created,
+	// which will not happen without at least one DeliveryToCypress.
+	image := extractBuiltinSidecarImageFromSpec(commonSpec, *instanceSpec)
+	for _, structuredLogger := range instanceSpec.StructuredLoggers {
+		if structuredLogger.CypressDelivery != nil && structuredLogger.CypressDelivery.Enabled {
+			return image
+		}
+	}
+	return ""
+}
+
+func (s *serverImpl) patchWithTimbertruck(statefulSet *appsv1.StatefulSet, structuredLoggers []ytv1.StructuredLoggerSpec) {
+	image := extractTimbertruckImageFromSpec(s.commonSpec, s.instanceSpec)
+	if image == "" {
+		return
+	}
+
+	logsDeliveryPath := extractLogsDeliveryPathFromSpec(s.commonSpec, s.instanceSpec)
+
+	workDir := ""
+	for _, locations := range s.instanceSpec.Locations {
+		if locations.LocationType == ytv1.LocationTypeLogs {
+			workDir = locations.Path
+			break
+		}
+	}
+	if workDir == "" {
+		return
+	}
+	timbertruckConfig := ytconfig.NewTimbertruckConfig(
+		structuredLoggers,
+		path.Join(workDir, consts.TimbertruckWorkDirName),
+		consts.GetServiceKebabCase(s.labeller.ComponentType),
+		workDir,
+		s.cfgen.GetHTTPProxiesAddress(s.ytsaurusSpec, consts.DefaultHTTPProxyRole),
+		logsDeliveryPath,
+	)
+
+	if timbertruckConfig == nil {
+		return
+	}
+
+	timbertruckInitScript, err := getTimbertruckInitScript(timbertruckConfig)
+	if err != nil {
+		return
+	}
+
+	statefulSet.Spec.Template.Spec.Containers = append(statefulSet.Spec.Template.Spec.Containers, corev1.Container{
+		Name:    consts.TimbertruckContainerName,
+		Image:   image,
+		Command: []string{"/bin/bash", "-c", timbertruckInitScript},
+		Env: []corev1.EnvVar{
+			{
+				Name: consts.TokenSecretKey,
+				ValueFrom: &corev1.EnvVarSource{
+					SecretKeyRef: &corev1.SecretKeySelector{
+						LocalObjectReference: corev1.LocalObjectReference{
+							Name: fmt.Sprintf("%s-secret", consts.TimbertruckUserName),
+						},
+						Key: consts.TokenSecretKey,
+					},
+				},
+			},
+			{
+				Name:  "YT_PROXY",
+				Value: s.cfgen.GetHTTPProxiesAddress(s.ytsaurusSpec, consts.DefaultHTTPProxyRole),
+			},
+		},
+		VolumeMounts: []corev1.VolumeMount{
+			{Name: path.Base(workDir), MountPath: workDir, ReadOnly: false},
+		},
+		ImagePullPolicy: corev1.PullAlways,
+	})
+}
+
```

Comment:
Trying to understand the need of timbertuck related-code in this common server component.
Is it because we can configure timbertruck structured logs for any ytsaurus component?